src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Output/index.jsx

```diff
@@ -1,286 +1,163 @@
-        if (this.shallowCompare(this, nextProps, nextState))
-            this.updateState(nextProps);
-    }
-    shallowEqual(objA, objB) {
-        if (objA === objB) {
-            return true;
-        }
-
-        if (typeof objA !== "object" || objA === null ||
-            typeof objB !== "object" || objB === null) {
-            return false;
-        }
-
-        let keysA = Object.keys(objA);
-        let keysB = Object.keys(objB);
 
-        if (keysA.length !== keysB.length) {
-            return false;
-        }
-
-        // Test for A's keys different from B.
-        let bHasOwnProperty = hasOwnProperty.bind(objB);
-        for (let i = 0; i < keysA.length; i++) {
-            if (typeof objA[keysA[i]] !== "function" || typeof objB[keysA[i]] !== "function") {
-                if (!bHasOwnProperty(keysA[i]) || objA[keysA[i]] !== objB[keysA[i]]) {
-                    return false;
-                }
-            }
-        }
-
-        return true;
-    }
-    shallowCompare(instance, nextProps, nextState) {
-        let result = (
-            !this.shallowEqual(instance.props, nextProps) ||
-            !this.shallowEqual(instance.state, nextState)
-        );
-        return result;
-    }
-
-    updateState(props) {
-        let { state } = this;
-        let self = this;
-        state.commandList = props.commandList;
-        state.output = props.output;
-        state.data = props.data;
-        state.paging = props.paging;
-        state.isHtml = props.isHtml;
-        state.isError = props.isError;
-        state.fieldOrder = props.fieldOrder;
-        state.reload = props.reload;
-        state.clearOutput = props.clearOutput;
-        state.style = props.style;
-        state.isHelp = props.isHelp;
-        state.name = props.name;
-        state.description = props.description;
-        state.options = props.options;
-        state.resultHtml = props.resultHtml;
-        state.error = props.error;
-        state.nextPageCommand = props.nextPageCommand;
-        this.setState({}, () => {
-            self.renderResults();
-        });
-    }
     renderResults() {
-        let { state } = this;
-        let { props } = this;
+        const { props } = this;
+
         props.IsPaging(false);
-        const style = state.style ? state.style : state.isError ? "error" : "ok";
-        let { fieldOrder } = state;
-        if (state.isHelp) {
-            if (state.commandList !== null && state.commandList.length > 0) {
-                this.renderCommands();
+        let { fieldOrder } = props;
+        if (props.isHelp) {
+            if (props.commandList !== null && props.commandList.length > 0) {
+                return this.renderCommands();
             } else {
-                this.renderHelp();
+                return this.renderHelp();
             }
-            return;
         }
         if ((typeof fieldOrder === "undefined" || !fieldOrder || fieldOrder.length === 0) && fieldOrder !== null) {
```

Comment:
Does this series of `ifs` look like something that could be split into smaller components?

Reply:
More specifically it looks like each of the `render` methods should be its own component

Reply:
Yes, it should be. Moreover both Input and Output components could be stateless functional components easily.
I propose to keep it as is for a transitional situation. Let's open a new issue for a second step refactoring in parallel with unit tests.

Reply:
Did you propose to do this refactoring later and then just did it? 😂